File: JAVA/binarySearch.java
// Binary Search in Java

class BinarySearch {

  int binarySearch(int array[], int x, int low, int high) {

    if (high >= low) {

      int mid = low + (high - low) / 2;

      // If found at mid, then return it

      if (array[mid] == x)

        return mid;

      // Search the left half

      if (array[mid] > x)

        return binarySearch(array, x, low, mid - 1);

      // Search the right half

      return binarySearch(array, x, mid + 1, high);

    }

    return -1;

  }

  public static void main(String args[]) {

    BinarySearch ob = new BinarySearch();

    int array[] = { 3, 4, 5, 6, 7, 8, 9 };

    int n = array.length;

    int x = 4;

    int result = ob.binarySearch(array, x, 0, n - 1);

    if (result == -1)

      System.out.println("Not found");

    else

      System.out.println("Element found at index " + result);

  }

}

File: JAVASCRIPT/binary_search.js
// binary search code in javascript

function binarySearch(sortedArray, key) {
    let start = 0;
    let end = sortedArray.length - 1;

    while (start <= end) {
        let middle = Math.floor((start + end) / 2);

        if (sortedArray[middle] === key) {
            // found the key
            return middle;
        } else if (sortedArray[middle] < key) {
            // continue searching to the right
            start = middle + 1;
        } else {
            // search searching to the left
            end = middle - 1;
        }
    }
    // key wasn't found
    return -1;
}

// array (array should be sorted)
let array = [11, 22, 33, 44, 55, 66, 77, 88, 99]

// key to be searched in array
let key = 44;

console.log(binarySearch(array, key))